extra/onnx.py

```diff
@@ -92,7 +94,7 @@ def run_onnx(inputs={}, debug=False):
       if inp.name in tensors: continue
       tmp=inp.type.optional_type.elem_type.tensor_type if inp.type.HasField("optional_type") else (inp.type.sequence_type.elem_type.tensor_type if inp.type.HasField("sequence_type") else inp.type.tensor_type)
       shape = shape_to_tuple(tmp.shape)
-      if len(shape) >= 1 and shape[0] == 0: shape = tuple([1]+list(shape[1:]))   # 1 batch size
+      if len(shape) >= 1: shape = tuple([x if x != 0 else 1 for x in shape])  # replace all dynamic dims with 1 for now
       if inp.name in inputs:
         if isinstance(inputs[inp.name], Tensor):
           input_tensors[inp.name] = inputs[inp.name]
```

Comment:
This fixes it for me:
```suggestion
      if inp.name in inputs:
        if isinstance(inputs[inp.name], Tensor):
          input_tensors[inp.name] = inputs[inp.name]
        else:
          input_tensors[inp.name] = Tensor(inputs[inp.name], requires_grad=False)
        input_shape = input_tensors[inp.name].shape
        if input_shape == (0,): raise NotImplementedError("empty tensors aren't supported in tinygrad")

        if len(shape) >= 1: shape = tuple([x if x != 0 else input_dim for x, input_dim in zip(shape, input_shape)]) # if input_shape is dynamic, replace with input_shape
```